.changeset/curly-rats-fold.md

```diff
@@ -0,0 +1,256 @@
+---
```

Comment:
This file content is what goes into the CHANGELOG.md and this is a bit too much and internal :)
See https://backstage.io/docs/getting-started/contributors/#writing-changesets for some examples

Reply:
I'll update the changeset to be more simple.

Reply:
I think most of it can be captured just as "Internal refactoring for performance in the service handlers" or something (at patch level), but the change to the response shape is so critical that I'd actually prefer if it were a separate pull request that called out super explicitly in its changeset, something like

```
**BREAKING**: Relations, as returned through the `relations` field of entities in the catalog REST API responses, no longer have the `target` property. This property has been removed from the type system since #9972 in March 2022, but was kept around in the response shape until now for backwards compatibility reasons. Please use the `targetRef` field which is a full standard entity ref, instead of `target`.
```

This calls it out very clearly. It could even have small "before" and "after" snippets showing the change of shape. Then it's also simple to revert in isolation.

Normally this would need a major version bump of the catalog. It makes me a bit nervous, and we're sure to get reports of people being affected. And the only benefit to people is slightly smaller response sizes. I'm still tempted to say go ahead.

Ping @backstage/catalog-maintainers @backstage/maintainers

Reply:
@freben I created a new [pr](https://github.com/backstage/backstage/pull/17457) for the breaking change.